src/Analysis/Ast/Impl/Analyzer/Handlers/LoopImportedVariableHandler.cs

```diff
@@ -0,0 +1,134 @@
+            }
+
+            if (!_asts.TryGetValue(key, out var ast)) {
+                return _cachedVariables.TryGetValue(key, out var variables)
+                    ? variables.Names
+                    : variableModule.GetMemberNames().Where(s => !s.StartsWithOrdinal("_"));
+            }
+
+            walker = WalkModule(module, ast);
+            return walker != null ? GetMemberNames(walker, variableModule) : variableModule.GetMemberNames();
+        }
+
+        public IVariable GetVariable(in PythonVariableModule variableModule, in string name) {
+            var module = variableModule.Module;
+            if (module == null || _isCanceled()) {
+                return default;
+            }
+
+            if (module.Analysis is DocumentAnalysis) {
+                return module.Analysis.GlobalScope.Variables[name];
+            }
+
+            var key = new AnalysisModuleKey(module);
+            if (_walkers.TryGetValue(key, out var walker)) {
+                return walker.Eval.GlobalScope?.Variables[name];
+            }
+
+            if (!_asts.TryGetValue(key, out var ast)) {
+                return _cachedVariables.TryGetValue(key, out var variables) ? variables[name] : default;
+            }
+
+            walker = WalkModule(module, ast);
+            var gs = walker != null ? walker.Eval.GlobalScope : module.Analysis.GlobalScope;
+            return gs?.Variables[name];
+        }
+
+        public void EnsureModule(in PythonVariableModule variableModule) {
+            var module = variableModule.Module;
+            if (module == null || _isCanceled()) {
+                return;
+            }
+
+            var key = new AnalysisModuleKey(module);
+            if (!_walkers.ContainsKey(key) && _asts.TryGetValue(key, out var ast)) {
+                WalkModule(module, ast);
+            }
+        }
+
+        public ModuleWalker WalkModule(IPythonModule module, PythonAst ast) {
+            var analyzer = _services.GetService<IPythonAnalyzer>();
+            
+            var analysis = analyzer.TryRestoreCachedAnalysis(module);
+            if(analysis != null) {
+                return null;
+            }
+
+            // If module has stub, make sure it is processed too.
+            if(module.Stub?.Analysis is EmptyAnalysis) {
+                WalkModule(module.Stub, module.GetAst());
+            }
+
+            var eval = new ExpressionEval(_services, module, ast);
+            var walker = new ModuleWalker(eval, this);
+
+            _walkers[new AnalysisModuleKey(module)] = walker;
+            ast.Walk(walker);
+            walker.Complete();
+
+            return walker;
+        }
+
+        private static IEnumerable<string> GetMemberNames(ModuleWalker walker, PythonVariableModule variableModule)
+            => walker.StarImportMemberNames ?? walker.GlobalScope.GetExportableVariableNames().Concat(variableModule.ChildrenNames).Distinct().Where(s => !s.StartsWithOrdinal("_"));
```

Comment:
Unless I'm misunderstanding this part of the code, all of the instances of `StartImportMemberNames` will need to be taken out and returned to the star import handling; using `__all__` for anything other than star imports isn't correct and the old code didn't behave this way.

Reply:
Resolving; this between the two implementations of `IImportedVariableHandler`, this is consistent.